### PR TITLE
Added phpstan to do static code analysis + fixed found bugs

### PR DESCRIPTION
**Pull request by mhlavac**
Hi David, thanks for amazing repo. I though I'll try out phpstan on it and found few minor bugs.

Added phpstan to do static code analysis + fixed found bugs:

* Set up PHPStan to run automatically via make
* Updated makefile to download composer.phar only once
* Fixed return value that should be null instead of void
* Fixed exception returned that should be concrete exception and not the interface provided by Doctrine.

### CODE REVIEW

File: src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticDriver.php [hunk 1]
```diff
@@ -97,7 +98,7 @@ public function convertException($message, DriverException $exception)
             return $this->underlyingDriver->convertException($message, $exception);
         }
 
-        return $exception;
+        return new Exception\DriverException($message, $exception);
```
Comment (dmaicher) [inline src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticDriver.php]: nice catch! This was wrong indeed :yum:

**Comment by dmaicher** [comment]
Thanks! Totally up for adding this to the build. Can you make sure the travis build passes?

We should only run phpstan for symfony 4 and php 7

**Comment by mhlavac** [comment]
@dmaicher thanks, I fixed it. Unfortunately I couldn't keep it as dependency inside composer, because it won't install on PHP <7.

For a tool wget might be just fine, if you drop support for php 5.x I can move it to composer.json

File: makefile [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
 	php composer.phar install --prefer-dist -o --dev
 
+.PHONY: test
 test:
 	vendor/bin/phpunit -c tests/ tests/
 
-build: init_composer test
+.PHONY: analyse
+analyse: phpstan.phar
+	php -r 'exit(PHP_VERSION_ID > 70000 ? 0 : 1);' && ./phpstan.phar analyse -c phpstan.neon -a vendor/autoload.php -l 7 src || echo "skipping analysis"
```
Comment (dmaicher) [inline makefile]: now the build will pass if phpstan fails, right? The `|| echo "skipping analysis"` will take precedence?

**Comment by mhlavac** [comment]
@dmaicher finally, thanks for great suggestion! :-)

**Comment by dmaicher** [comment]
:+1: thanks @mhlavac

**Pull request merged**